pkg/config/config_test.go

```diff
@@ -111,11 +111,17 @@ type receiverTestConfig struct {
 	WontFixResolution string `yaml:"wont_fix_resolution,omitempty"`
 	AddGroupLabels    bool   `yaml:"add_group_labels,omitempty"`
 
+	AutoResolve *testAutoResolve `yaml:"auto_resolve" json:"auto_resolve"`
+
 	// TODO(rporres): Add support for these.
 	// Fields            map[string]interface{} `yaml:"fields,omitempty"`
 	// Components        []string               `yaml:"components,omitempty"`
 }
 
+type testAutoResolve struct {
```

Comment:
nit: missing docstring

Reply:
This is similar to how `testConfig` and `receiverTestConfig` are being used. I followed the existing pattern here.

Reply:
`receiverTestConfig` and `testConfig` are there because they are and they have to be different to the original structs. If the original structs could have been reused, they would have been.

But there's nothing in `testAutoResolve` that is different to the original `AutoResolve`, hence it's better to reuse it.